Examples/WindowsFormsApplication/readme.md

```diff
@@ -0,0 +1,127 @@
+- Client ID: this is the ID of the application we just registered
+- Tenant ID: the ID of the tenant that we are signing into
+- Nfield Public API Application ID: The ID of the Nfield Public API
+
+These values will have to be added to `appsettings.json`.
+Besides that the name of the domain should be added to this config.
+It will be added as a header to each API call.
+Luckily the Nfield SDK will take care of that for us.
+
+## Application
+
+Now that we have everything setup for our application, let's look at what we needed to do to use Azure AD for authentication.
+The interaction between the application and Nfield is encapsulated in the NField class.
+
+This class uses the MSAL library to interact with Azure AD.
+It can be installed by installing the [Microsoft.Identity.Client](https://www.nuget.org/packages/Microsoft.Identity.Client/) NuGet package.
+
+First we need to initialize a client:
+```csharp
+var client = PublicClientApplicationBuilder.Create(ApplicationConfiguration.Current.ClientId)
+    .WithRedirectUri("http://localhost")
+    .WithAuthority(AzureCloudInstance.AzurePublic, ApplicationConfiguration.Current.Tenant)
+    .Build();
+```
+The reference to the redirect URI to `http://localhost` allows the application to retrieve the token that is returned by Azure AD from the browser.
+
+The method `AuthenticateAsync` contains the logic to acquire an access token from Azure AD.
+```csharp
+public async Task<AuthenticationResult> AuthenticateAsync(Action<string> statusCallback)
+{
+    var accounts = await ClientApp.GetAccountsAsync();
+    var account = accounts.FirstOrDefault();
+
+    try
+    {
+        return await ClientApp.AcquireTokenSilent(Scopes, account).ExecuteAsync();
+    }
+    catch (MsalUiRequiredException mure)
+    {
+        statusCallback(mure.Message);
+    }
+
+    var authResult = await ClientApp.AcquireTokenInteractive(Scopes)
+        .WithAccount(account)
+        .WithPrompt(Prompt.SelectAccount)
+        .ExecuteAsync();
+
+    return authResult;
+}
+```
+First we try to get a token without user interaction (in case we have a valid refresh token).
+If that fails, the code falls back to a method that will open the system browser and allows the user to select the account they want to use to login.
+
+Please take note of the method that determines the name of the scopes.
+```csharp
+private static string CreateScope(string scope) => $"{ApplicationConfiguration.Current.NfieldApiApplicationId}/{scope}";
+```
+Nfield only supports scopes based on the Nield Public API application ID.
+The name as presented in the permissions UI in the Azure portal cannot not be used.
+
+Now after launching the application and logging in for the first time the user is asked to consent giving access to the application. For the sample application the screen will look like this:
+
+![Consent](./images/Consent.png)
+
+After giving consent the user should be presented with a screen that shows the surveys in the domain.
```

Comment:
It would be nice to also link to some formal MS docs if people would like to know more (especially on the permission part/app config).

Reply:
I have sprinkled some nice links throughout the document.